metricbeat/module/munin/munin.go

```diff
@@ -0,0 +1,109 @@
+package munin
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"net"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/joeshaw/multierror"
+	"github.com/pkg/errors"
+
+	"github.com/elastic/beats/libbeat/common"
+)
+
+const (
+	unknownValue = "U"
+)
+
+// Node connection
+type Node struct {
+	conn net.Conn
+
+	writer io.Writer
+	reader *bufio.Reader
+}
+
+// Connect with a munin node
+func Connect(address string, timeout time.Duration) (*Node, error) {
+	conn, err := net.DialTimeout("tcp", address, timeout)
+	if err != nil {
+		return nil, err
+	}
+	n := &Node{conn: conn,
+		writer: conn,
+		reader: bufio.NewReader(conn),
+	}
+	// Cosume and ignore first line returned by munin, it is a comment
+	// about the node
+	bufio.NewScanner(n.reader).Scan()
```

Comment:
Should we check for the return value here? In case of `false` we should probably check the `Err` method for the error.

Reply:
Yes, returning `scanner.Error()`.